### PR TITLE
Update @vue/cli to V5

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 101280275

## Summary

Updates Vue CLI to latest version along with its deps. Uses native Webpack fallback folder mechanisms, for our theming setup.

## Dependencies

- https://github.com/apple/swift-docc-render/pull/440

## Testing

Make sure everything builds without errors

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@marinaaisa @mportiz08 @hqhhuang please review this as well, so we can update critical deps

File: .eslintrc.js [hunk 1]
```diff
@@ -35,6 +35,14 @@ module.exports = {
...
     'template-curly-spacing': 'off',
     'vue/experimental-script-setup-vars': 'off',
+    'vue/multi-word-component-names': 'off',
+    'function-paren-newline': ['error', 'consistent'],
+    'function-call-argument-newline': 'off',
+    'vuejs-accessibility/form-control-has-label': 'off',
+    'vuejs-accessibility/label-has-for': 'off',
+    'vuejs-accessibility/aria-role': 'off',
+    'vuejs-accessibility/click-events-have-key-events': 'off',
+    'vuejs-accessibility/anchor-has-content': 'off',
```
Comment (dobromir-hristov) [inline .eslintrc.js:45]: These are all new rules which I disabled, so we dont change too much on our code.

File: package.json [hunk 2]
```diff
@@ -27,23 +27,27 @@
     "intersection-observer": "0.12.0",
     "portal-vue": "2.1.7",
     "vue-i18n": "^8.28.2",
-    "vue-virtual-scroller": "^1.1.2",
-    "webpack-theme-resolver-plugin": "3.0.0"
```
Comment (marinaaisa) [inline package.json:31]: Can we delete the info about webpack-theme-resolver-plugin in the README as well?
https://github.com/dobromir-hristov/swift-docc-render/blob/dhristov/update-vue-cli/NOTICE.txt#L78

**Review by marinaaisa — APPROVED** [review]
I've been testing it out and it works great!

I left a very minor comments! 👏

**Review by hqhhuang — APPROVED** [review]
I was getting this error earlier:

>  ERROR  ValidationError: Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
         - options has an unknown property 'before'. These properties are valid:
           object { allowedHosts?, bonjour?, client?, compress?, devMiddleware?, headers?, historyApiFallback?, host?, hot?, http2?, https?, ipc?, liveReload?, magicHtml?, onAfterSetupMiddleware?, onBeforeSetupMiddleware?, onListening?, open?, port?, proxy?, server?, setupExitSignals?, setupMiddlewares?, static?, watchFiles?, webSocketServer? }
           
Somehow got it to work after re-doing my `.env.development.local` file. 
Everything works as expected now!

**Comment by dobromir-hristov** [comment]
@hqhhuang @marinaaisa great catch you two. That was caused by the use of the deprecated `before` config  in Webpack 5 devserver.

**Pull request merged**